### PR TITLE
Connect React and Redux

### PR DESCRIPTION
**Pull request by jlvFlores**
For this PR, I:
- Removed states from React files and replaced received values with those from Redux store.
- Added uuid dependency to create unique ID's.
- Removed unnecessary keys and renamed genre to category.
- Added a remove book button, and added styles to it.
- Fixed add book reducer.

# 👉 Important comment for Future CR: 👈 👀
It was necessary to add an esLint disable comment on `booksSlice.js`, this is because the [Redux documentation](https://redux-toolkit.js.org/api/createSlice) recommends directly changing the state parameter, however esLint currently conflicts with this because it thinks the parameter is being reassigned, even though createSlice uses ImmerJs to take care of this on the back, so even if it looks like mutable code, immutable logic is happening thanks to Redux and immerJS.

P.S. I've added this comment because of a previous comment in my previous [PR](https://github.com/jlvFlores/Bookstore/pull/2).

### CODE REVIEW

**Review by Reem-lab — COMMENTED** [review]
Hi @jlvFlores, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that have a live demo section in your README file and to have a professional README file you should remove this section till you have your app deployed so you can just add the live demo section in case you have deployed link for your project is present. .✅❗💯🤩✨✨


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,9 +1,29 @@
 /* eslint-disable no-param-reassign */
```
Comment (didierganthier) [inline src/redux/books/booksSlice.js]: Well done overall 💯 I read your comment on the previous review and this one, but while doing some research, I found a workaround allowing you to do it with all `eslint` rules enabled, the code is still working 🔍 . First, you have to understand where the error is coming from. This ESLint error is caused by the fact that you are trying to modify the state object directly inside the reducer functions 👀 , which is generally not recommended. In Redux, the state is considered immutable and any updates to it should be done in an immutable way.

To fix this error, you can create a new state object based on the previous state and the changes you want to make. In your `addBook` reducer, you can create a new array that includes the previous books and the new book you want to add, like this:
```
addBook: (state, action) => {
  const newBook = {
    item_id: uuidv4(),
    title: action.payload.title,
    author: action.payload.author,
    category: action.payload.category,
  };
  return { ...state, books: [...state.books, newBook] };
},
```
Similarly, in your `removeBook` reducer, you can create a new array that excludes the book you want to remove, like this:
```
removeBook: (state, action) => {
  const bookId = action.payload;
  return { ...state, books: state.books.filter((book) => book.item_id !== bookId) };
},
```
By creating a new state object this way, you are ensuring that the original state object remains unchanged, and the changes you make are done in an immutable way, which is recommended in Redux. 🤓

**Review by didierganthier — CHANGES_REQUESTED** [review]
Hi @jlvFlores ,

🎉 Good job so far! You're making great progress on this project, but there are still a few areas that need attention before we can move on to the next step.



## Highlights
- Descriptive PR ✔️ 
- Linters passing ✔️ 
- Correct Gitflow ✔️ 

## Required Changes 🛠
- Please review the comments under the review for specific changes that still need to be made.

## Optional Suggestions 💡
- While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code.

🚀 Keep pushing forward, you're almost there! If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

**Please note**
- To avoid confusion and delays, please do not open a new pull request for re-reviews. Use the same pull request submitted for the first review, unless otherwise instructed.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Cheers and happy coding! 👏👏👏

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (jlvFlores) [reply src/redux/books/booksSlice.js]: Wow, thank you, @didierganthier I can't believe it was that simple, I truly appreciate the commitment you've put into finding a working solution, I owe you one. 💪

**Review by didierganthier — APPROVED** [review]
Hi @jlvFlores ,

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀



## Highlights
- Descriptive PR ✔️ 
- Linters passing ✔️ 
- Correct GitHub flow ✔️ 

## Optional Suggestions 💡
- N/A

Cheers and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Once again, congratulations on completing this project! 🎉

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (didierganthier) [reply src/redux/books/booksSlice.js]: Anytime man 😉 🚀

**Pull request merged**